services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationData.kt

```diff
@@ -107,6 +111,29 @@ data class CableReportApplicationData(
 
     fun findOrderer(): Contact? =
         customersWithContacts().flatMap { it.contacts }.find { it.orderer }
+
+    fun toHakemusData(applicationEntity: ApplicationEntity): HakemusData =
+        JohtoselvityshakemusData(
+            name = name,
+            postalAddress = postalAddress,
+            constructionWork = constructionWork,
+            maintenanceWork = maintenanceWork,
+            propertyConnectivity = propertyConnectivity,
+            emergencyWork = emergencyWork,
+            rockExcavation = rockExcavation,
+            workDescription = workDescription,
+            startTime = startTime,
+            endTime = endTime,
+            areas = areas,
+            customerWithContacts =
+                applicationEntity.yhteystiedot[ApplicationContactType.HAKIJA]?.toDomain(),
```

Comment:
These need a Hibernate Session (transaction) to work. One cannot access ApplicationEntity.yhteystiedot otherwise. Or at least so I figured after banging my head to that tree for too long.

Reply:
But now that I check the code further, I see that this is actually used in a `@Transactional HakemusService.getHakemuksetForKayttaja`. So, no problems. But one cannot use this in tests (without session/transaction).

Reply:
In production code, we shouldn't be passing entities over transaction boundaries.

In integration tests, the correct way to get a domain object is to call the service method that loads it from the DB.

In unit tests, there is no DB or entity manager, so there are no lazily loaded values.

Reply:
That being said, ApplicationData is not an entity, so I don't think it should be taking entities as parameters, either. The border between entities and domain objects is not clear.

I refactored this to take a map of Hankeyhteystieto domain objects instead.